### PR TITLE
[TT-5262] add dialCtxFn in gw to test.

### PR DESCRIPTION
**Pull request by jeffy-mathew**
fix: upstream mTLS flaky tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4176/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2633905108)

File: gateway/reverse_proxy.go [hunk 1]
```diff
@@ -403,13 +403,19 @@ func (p *ReverseProxy) defaultTransport(dialerTimeout float64) *http.Transport {
...
+	transport := &http.Transport{
 		DialContext:           dialContextFunc,
 		MaxIdleConns:          p.Gw.GetConfig().MaxIdleConns,
 		MaxIdleConnsPerHost:   p.Gw.GetConfig().MaxIdleConnsPerHost, // default is 100
 		ResponseHeaderTimeout: time.Duration(dialerTimeout) * time.Second,
 		TLSHandshakeTimeout:   10 * time.Second,
 	}
+
+	if p.Gw.dialCtxFn != nil {
+		transport.DialContext = p.Gw.dialCtxFn
```
Comment (titpetric) [inline gateway/reverse_proxy.go:415]: Would you not override this sooner (L407, L403...?)

**Review by titpetric — APPROVED** [review]

File: gateway/reverse_proxy.go (hunk 1 above)
Comment (jeffy-mathew) [reply gateway/reverse_proxy.go:415]: yep, that'd do :)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4176/merge
Commit: 42b14b2865d583ad2cfde01d00c2b2598a41e2ef 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2635262007)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4176/merge
Commit: d5db76c676a2fb07e6b574ce102561e5d126db07 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2635916150)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4176/merge
Commit: bb47c66a6cfaa32ceace5574b1a05449c7a9f106 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2636104410)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4176/merge
Commit: 4a7d7715ae6037fa35bb72c4701766989ea961b2 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2688336656)

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4176/merge
Commit: 7da85cf3d95ff65a37fed246655b2a873bb110d9 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2717264214)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4176/merge
Commit: e19dd5ddbe1eef9bdc66c2d010cc735913b1eea9 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2717623391)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 66942b34632f6733497763fdeecf593f1d3de96f [TT-5262] add dialCtxFn in gw to test. (#4176)

* add dialCtxFn in gw to test.
[changelog]
fix: upstream mTLS flaky tests.
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2737999328)

**Pull request merged**